crates/wast/src/names.rs

```diff
@@ -51,11 +51,13 @@ impl<'a> Namespace<'a> {
     }
 
     pub fn register_specific(&mut self, name: Id<'a>, index: u32, desc: &str) -> Result<(), Error> {
-        if let Some(_prev) = self.names.insert(name, index) {
-            return Err(Error::new(
-                name.span(),
-                format!("duplicate identifier for {}", desc),
-            ));
+        if let Some(prev) = self.names.insert(name, index) {
```

Comment:
I think it would be best to leave out this change since, as you commented on the original issue, this isn't a complete fix. I find that this makes the state of things more confusing since it appears to sometimes work but still sometimes not work.

Reply:
Reverting. Will start a new PR with a proper fix.